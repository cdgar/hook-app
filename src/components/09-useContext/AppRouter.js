import React from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route
  } from 'react-router-dom';
import NavBar from './NavBar';
import AboutScreen from './AboutScreen';
import HomeScreen from './HomeScreen';
import LoginScreen from './LoginScreen';
import NotFountScreen from './NotFountScreen';

const AppRouter = () => {
    return (
        <Router>
            <NavBar/>
            <div className="container">
                <Switch>
                    <Route exact path="/" component={HomeScreen} />
                    <Route exact path="/about" component={AboutScreen} />
                    <Route exact path="/login" component={LoginScreen} />
                    <Route exact path="" component={NotFountScreen} />
                </Switch>
            </div>
        </Router>
    )
}

export default AppRouter
