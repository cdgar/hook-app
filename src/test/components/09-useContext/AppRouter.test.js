import { mount } from 'enzyme';
import React from 'react';
import AppRouter from '../../../components/09-useContext/AppRouter';
import UserContext from '../../../components/09-useContext/UserContext';
import { userMock } from '../../fixtures/userMock';

describe('AppRouter.js', () => {

    const wrapper = mount(
        <UserContext.Provider value={{user: userMock}}>
            <AppRouter />
        </UserContext.Provider>
    );

    test('should AppRouter', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
})
