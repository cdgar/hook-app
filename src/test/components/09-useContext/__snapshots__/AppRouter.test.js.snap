// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRouter.js should AppRouter 1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light"
  >
    <div
      className="container-fluid"
    >
      <a
        aria-current="page"
        className="navbar-brand active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        useContext
      </a>
      <div
        className="collapse navbar-collapse"
        id="navbarNavAltMarkup"
      >
        <div
          className="navbar-nav"
        >
          <a
            aria-current="page"
            className="nav-link active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </div>
      </div>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div>
      <h1>
        HomeScreen
      </h1>
      <hr />
      <pre>
        {
   "name": "nameXD",
   "email": "email@email.com"
}
      </pre>
    </div>
  </div>,
]
`;
